src/mpid/ch4/shm/posix/posix_init.c

```diff
@@ -264,6 +264,7 @@ int MPIDI_POSIX_coll_finalize(void)
     /* Destroy the shared counter which was used to track the amount of shared memory created
      * per node for intra-node collectives */
     mpi_errno = MPIDU_Init_shm_free(MPIDI_POSIX_global.shm_ptr);
+    MPIDI_POSIX_shm_limit_counter = NULL;
```

Comment:
Having to consider the order of "init" and "finalize" to understand this need of checking `MPIDI_POSIX_shm_limit_counter` is a complexity burden that I like to avoid. How about this --
let's set `MPIDI_POSIX_shm_limit_counter = &dummy_shm_limit_counter;` where `dummy_shm_limit_counter` is a static variable. Then all the other changes in this commit become unnecessary, right?

Reply:
Add  comment to explain the reason if you do so.